### PR TITLE
Add analytics tracking to unsupported browser page

### PR DESCRIPTION
**Pull request by msmanoj**
## Type

- [ ] Bug fix
- [ ] New feature
- [x] Improvement to existing feature

## Related JIRA Issue(s)
https://www.ebi.ac.uk/panda/jira/browse/ENSWBSITES-1464

## Description
- added a call to the `trackPageView` method provided by analytics-service in the unsupported browser page

## Deployment URL
http://unsupported-browser-tracking.review.ensembl.org/unsupported-browser

## Views affected
Unsupported browser page

### CODE REVIEW

**Comment by msmanoj** [comment]
Hi @azangru - which file do you think we can add this script tag to?

**Comment by azangru** [comment]
https://github.com/Ensembl/ensembl-client/blob/dev/src/server/routes/unsupportedBrowserRouter.tsx#L43

File: src/server/routes/unsupportedBrowserRouter.tsx [hunk 1]
```diff
@@ -49,6 +49,17 @@ const unsupportedBrowserRouter = (_: Request, res: Response) => {
       <title>Unsupported browser</title>
       <meta name="description" content="Your browser is not supported"></meta>
       <meta name="viewport" content="width=device-width, initial-scale=1.0">
+      <script>
+      if('${process.env.REPORT_ANALYTICS}'.toLowerCase() === 'true'){
```
Comment (azangru) [inline src/server/routes/unsupportedBrowserRouter.tsx:53]: We could go one step further and make the whole addition of the script to the payload conditional on `process.env.REPORT_ANALYTICS`.

File: src/server/routes/unsupportedBrowserRouter.tsx [hunk 2]
```diff
@@ -39,6 +39,20 @@ const unsupportedBrowserRouter = (_: Request, res: Response) => {
...
+    if('${process.env.REPORT_ANALYTICS}'.toLowerCase() === 'true'){
+      (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
+      (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
+      m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
+      })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');
+
+      ga('create', '${process.env.GOOGLE_ANALYTICS_KEY}', 'auto');
+      ga('send', 'pageview');
+    }
+    </script>`;
```
Comment (azangru) [inline src/server/routes/unsupportedBrowserRouter.tsx:55]: I'm confused why there are two checks of `'${process.env.REPORT_ANALYTICS}'.toLowerCase() === 'true'` here.

I'd also move this script out out of the router function for readability.

```js
const getAnalyticsScript = () =>
  `<script>
    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
    (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
    m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
    })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');
    ga('create', '${process.env.GOOGLE_ANALYTICS_KEY}', 'auto');
    ga('send', 'pageview');
  </script>`;


// in the unsupportedBrowserRouter, inside the html template

${'${process.env.REPORT_ANALYTICS}'.toLowerCase() === 'true' && getAnalyticsScript()}
```

**Review by azangru — APPROVED** [review]
Turns out we forgot that we were not dealing with JSX 😄 Good thing I tested locally.

image

I've pushed a commit with a fix.

**Pull request merged**